### DISCUSSION TITLE
Failed to load plugin '@next/next' declared in '.eslintrc.js'

### DISCUSSION THREAD

**Discussion by aafnnp** [vercel/next.js#38220] [Help] [answered]
```bash
error - ESLint: Failed to load plugin '@next/next' declared in '.eslintrc.js': Unexpected token '.' Referenced from: 
info  - Creating an optimized production build...
warn  - Found lockfile missing swc dependencies, run next locally to automatically patch
```

**Comment by icyJoseph** [comment, marked as answer]
mmm what is @next/next? the next plugin is called `eslint-config-next`, and we typically extend `next/core-web-vitals` in the eslint config.

Would you mind sharing more? `package.json` and `.eslintrc.js`, also file structure if possible.

**Reply by aafnnp** [reply]
```js
extends: [
		'eslint:recommended',
		'plugin:@next/next/recommended',
		'plugin:@typescript-eslint/recommended',
		'prettier', 
	],
```

**Reply by icyJoseph** [reply]
What next.js version are you using? 12.2.0, works fine

**Reply by aafnnp** [reply]
after upgrade to 12.2.0,this problem become.

**Reply by icyJoseph** [reply]
What version of eslint do you have?

**Reply by aafnnp** [reply]
7.32.0

**Reply by icyJoseph** [reply]
Mmm I downgraded from 8.13.0 to 7.32.0 and it still worked. Not sure I did it correctly.

I am running out of ideas here. Do you have a repository to take a look at?

**Reply by aafnnp** [reply]
upgrade node to v16,fixed.
company's project

**Reply by aafnnp** [reply]
thx

**Reply by icyJoseph** [reply]
Aha! What node version did you have before? 14? It must have been new syntax... Good catch!

**Reply by aafnnp** [reply]
yes.14

**Reply by icyJoseph** [reply]
Alright, please comment what you did on a separate reply and label it is as answered! Good job!

**Reply by tofuwu-dev** [reply]
Knew something new!

**Comment by mgouault** [comment]
Had the same error
```
yarn run v1.22.17
$ next lint
Failed to load plugin '@next/next' declared in '.eslintrc.json » eslint-config-next/core-web-vitals » /Users/mgouault/website/node_modules/eslint-config-next/index.js': Unexpected token '.'
Referenced from: /Users/mgouault/website/node_modules/eslint-config-next/index.js
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
Upgraded from node 12.22 to node 14.19 then reinstalled node_modules and it works

**Reply by stefaneidelloth** [reply]
Thank you. That solved the issue for me, too. Also see https://stackoverflow.com/questions/73036716/failed-to-load-plugin-next-next-when-running-npm-run-lint-in-gitlab-ci/73063116#73063116

**Comment by nimeshmaharjan1** [comment]
I am getting this error in neovim prettier is working fine. I am using null-ls. This is just a starter template from pnpx create-next-app -ts-

![image](https://user-images.githubusercontent.com/86953478/230541472-cbe6b4fb-c0c6-4fcb-9667-7fe1577b1bfa.png)

and this is the eslintrc.json file

![image](https://user-images.githubusercontent.com/86953478/230541583-8327737b-e3f1-494d-9b43-4d34ab74ff47.png)

**Reply by medromo** [reply]
Hi, were you able to fix this issue? Im having the same problem

**Reply by sjdonado** [reply]
I fixed it by using these versions:
```json
    "eslint-config-next": "^12.1.0",
    "eslint-plugin-react": "^7.28.0",
    "eslint-plugin-react-hooks": "^4.3.0",
```
then I restarted the eslint daemon `eslint_d restart`

**Reply by Nhahan** [reply]
this worked for me 😄

**Comment by nimeshmaharjan1** [comment]
Degrade next version I'm just using the pages for now bc the app gives me
so many errors and don't even know from where and why

**Reply by medromo** [reply]
Thanks, I appreciate the response

**Comment by nanxiaobei** [comment]
it's `@next/eslint-plugin-next`.

`pnpm add @next/eslint-plugin-next -D` fixed it

**Reply by yusukii-wang** [reply]
thx, that works for me

**Comment by Tordek** [comment]
In my case this was caused by two versions of typescript being installed in a monorepo project. The root version was 4.x and the child was 5.x. Manually installing typescript@^5.0.0 in the base project fixed the issue.
